### PR TITLE
WIP search function

### PR DESCRIPTION
**Pull request by SalSoloman**

### CODE REVIEW

File: routes/api.js [hunk 1]
```diff
@@ -33,5 +33,17 @@ app.post( '/', function (req, res) {
     })
 })
 
+app.get( '/search', function (req, res) {
+  const term = req.query.search
+  console.log(term)
```
Comment (Anasauce) [inline routes/api.js:47]: this is WIP PR but prob want to take out console.log before merging

File: routes/api.js [hunk 2]
```diff
@@ -33,5 +33,17 @@ app.post( '/', function (req, res) {
...
+  Books.search(term)
+  .then(results => {
+    res.status(200).json({data: results})
```
Comment (Anasauce) [inline routes/api.js:50]: just curious, why are you sending json? You can send it as an object that you can easily access in pug and also, add a redirect in your response 😸

File: views/navbar.pug [hunk 3]
```diff
@@ -8,6 +8,6 @@ nav.navbar.navbar-toggleable-md.navbar-light.bg-faded
         a.nav-link(href='/')
           | Home
           span.sr-only (current)
-    form.form-inline.my-2.my-lg-0
-      input.form-control.mr-sm-2(type='text', placeholder='Search')
-      button.btn.btn-outline-success.my-2.my-sm-0(type='submit') Search
+    form.form-inline.my-2.my-lg-0(method='get', action='/api/search')
```
Comment (Anasauce) [inline views/navbar.pug:11]: do you have your backend and frontend running on different servers? If not, you shouldn't have to use /api....just /search

File: routes/api.js (hunk 2 above)
Comment (FrankieFabuloso) [reply routes/api.js:50]: Were not using express's rendering for view templating. Our endpoints return json to an ajax call that then creates the DOM on success with the returned data.

Comment (FrankieFabuloso) [reply routes/api.js:50]: Were doing this cause I like having a hard life with unnecessary problems. ;)

File: views/navbar.pug (hunk 3 above)
Comment (FrankieFabuloso) [reply views/navbar.pug:11]: Yeah we do. Our 'index' routes just handle serving the empty html page. And our end points in api serve only data.

**Pull request merged**